work/v1/types.go

```diff
@@ -66,6 +70,65 @@ type DeleteOption struct {
 	SelectivelyOrphan *SelectivelyOrphan `json:"selectivelyOrphans,omitempty"`
 }
 
+// ManifestConfigOption represents the configurations of a manifest defined in workload field.
+type ManifestConfigOption struct {
+	// ResourceIdentifier represents the group, resource, name and namespace of a resoure.
+	// +kubebuilder:validation:Required
+	// +required
+	ResourceIdentifier ResourceIdentifier `json:"resourceIdentifier"`
+
+	// FeedbackRules defines what resource status field should be returned.
+	// +kubebuilder:validation:Required
+	// +required
+	FeedbackRules []FeedbackRule `json:"feedbackRule"`
+}
+
+type FeedbackRule struct {
+	// Type defines the option of how status can be returned.
+	// It can be jsonPaths only or wellKnownStatus.
+	// If the type is jsonPaths, user should specify the jsonPaths field
+	// If the type is wellKnownStatus, certain commont fields of status will be reported,
+	// including replicas, readyReplicas, and availableReplicas. And if these status fields
+	// do not exist, no values will be reported.
+	// +kubebuilder:validation:Required
+	// +required
+	Type FeedBackType `json:"type"`
+
+	// JsonPaths defines the json path under status field to be synced.
+	// +optional
+	JsonPaths []JsonPath `json:"jsonPaths,omitempty"`
```

Comment:
Something to consider in API evolution: do you want a way to define WellKnownStatus for particular resources so that someone's custom type can have well known status fields to avoid subtely different names?

Reply:
Yes, I think we could have another cluster scoped API to define the list of status field for custom type.